### ISSUE TITLE
ContractParameter is not convenient to use in InvocationResult

### ISSUE THREAD

**Issue by chenquanyu** [neow3j/neow3j#90]
When I want to call the testInvoke() to get balance of a token, the stack maybe retrurn ByteArray or BigInteger. It will be more convenient if neow3j can convert it to the type users wanted. In C#, the ContractParameter can be converted to StackItem.

```
ContractInvocation invoc = new ContractInvocation.Builder(neow3j)
                .contractScriptHash(contractHash)
                .function("balanceOf")
                // Add parameters in the correct order as they are expected by the contract.
                .parameter(ContractParameter.byteArray(ScriptHash.fromAddress(addr).toArray()))
                //.networkFee(new BigDecimal("0.1"))
                .build();

        InvocationResult result = invoc.testInvoke();
        byte[] value = (byte[]) result.getStack().get(0).getValue();
        BigDecimal balance = Numeric.fromFixed8ToDecimal(Numeric.toBigInt(ArrayUtils.reverseArray(value)));
```
```

        private static StackItem ToStackItem(ContractParameter parameter, List<Tuple<StackItem, ContractParameter>> context)
        {
            StackItem stackItem = null;
            switch (parameter.Type)
            {
                case ContractParameterType.Array:
                    if (context is null)
                        context = new List<Tuple<StackItem, ContractParameter>>();
                    else
                        stackItem = context.FirstOrDefault(p => ReferenceEquals(p.Item2, parameter))?.Item1;
                    if (stackItem is null)
                    {
                        stackItem = ((IList<ContractParameter>)parameter.Value).Select(p => ToStackItem(p, context)).ToList();
                        context.Add(new Tuple<StackItem, ContractParameter>(stackItem, parameter));
                    }
                    break;
                case ContractParameterType.Map:
                    if (context is null)
                        context = new List<Tuple<StackItem, ContractParameter>>();
                    else
                        stackItem = context.FirstOrDefault(p => ReferenceEquals(p.Item2, parameter))?.Item1;
                    if (stackItem is null)
                    {
                        stackItem = new Map(((IList<KeyValuePair<ContractParameter, ContractParameter>>)parameter.Value).ToDictionary(p => ToStackItem(p.Key, context), p => ToStackItem(p.Value, context)));
                        context.Add(new Tuple<StackItem, ContractParameter>(stackItem, parameter));
                    }
                    break;
                case ContractParameterType.Boolean:
                    stackItem = (bool)parameter.Value;
                    break;
                case ContractParameterType.ByteArray:
                case ContractParameterType.Signature:
                    stackItem = (byte[])parameter.Value;
                    break;
                case ContractParameterType.Integer:
                    stackItem = (BigInteger)parameter.Value;
                    break;
                case ContractParameterType.Hash160:
                    stackItem = ((UInt160)parameter.Value).ToArray();
                    break;
                case ContractParameterType.Hash256:
                    stackItem = ((UInt256)parameter.Value).ToArray();
                    break;
                case ContractParameterType.PublicKey:
                    stackItem = ((ECPoint)parameter.Value).EncodePoint(true);
                    break;
                case ContractParameterType.String:
                    stackItem = (string)parameter.Value;
                    break;
                case ContractParameterType.InteropInterface:
                    break;
                default:
                    throw new ArgumentException($"ContractParameterType({parameter.Type}) is not supported to StackItem.");
            }
            return stackItem;
        }
```

**Comment by claudemiller** [comment]
@nimmortal brought this topic up and started an implementation of a `StackItem` class that is now finished and merged in #91. The `ContractParameter`s are now replaced by `StackItem`s where it makes sense, e.g. in the `InvocationResult`. You can cast a `StackItem` to its appropriate subtype via convenience methods as follows.

```
StackItem item = invocationResult.getStack().get(0);
if (item.getType() == StackItemType.MAP) {
    MapStackItem mapItem = item.asMap();
    // do stuff
}
```

**Comment by claudemiller** [comment]
Waiting for release of StackItem feature.

**Comment by claudemiller** [comment]
Resolved in release 2.3.0
